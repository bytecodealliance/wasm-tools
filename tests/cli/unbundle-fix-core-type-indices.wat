;; RUN: component unbundle --module-dir %tmpdir --wat % --threshold 0

(component $C
  (core type $m (module))
  (import "x" (core module (type $m)))

  (component
  )

  (core module
    (func $f)
    (export "x" (func $f))
  )
)

