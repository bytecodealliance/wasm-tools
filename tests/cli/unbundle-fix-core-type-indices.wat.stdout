(component $C
  (core type (;0;) (func))
  (core type (;1;)
    (module
      (alias outer $C 0 (type (;0;)))
      (export "x" (func (type 0)))
    )
  )
  (import "unbundled-module0" (core module (;0;) (type 1)))
  (core type $m (;2;)
    (module)
  )
  (import "x" (core module (;1;) (type $m)))
  (component (;0;))
  (alias outer $C 0 (core module (;2;)))
)
